dashboard/app/controllers/script_levels_controller.rb

```diff
@@ -293,7 +293,9 @@ def present_level
 
     if @level.try(:pages)
       puzzle_page = params[:puzzle_page] || 1
-      @pages = [@level.pages[puzzle_page.to_i - 1]]
+      page = @level.pages[puzzle_page.to_i - 1]
+      raise ActiveRecord::RecordNotFound if page.nil?
+      @pages = [page]
```

Comment:
Interesting.  So raising an exception causes a 404?

It does seem somewhat reasonable to check for the failure case first, before setting page.